File: leetcode/69/Sqrt.js
var Sqrt = function(x) {
    let l = 0;
    let r = x;
    
    while(l <= r) {
      const mid = Math.floor((l + r) / 2);
      
      if (mid**2 === x) {
        return mid;
      } else if (mid**2 > x) {
        r = mid - 1;
      } else {
        l = mid + 1;
      }
    }
    
    return r;
  };

  console.log(Sqrt(10));

File: leetcode/69/Sqrt.java
class Solution {
    public int mySqrt(int x) {
        int left = 0;
        int right = x;

        while (left <= right) {
            int mid = (left + right) >> 1;
            double mul = Math.pow(mid, 2);
            if (mul == x) {
                return mid;
            } else if (mul > x) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return right;
    }
}